### PR TITLE
Update release process doc and checklist

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Direct links to the rendered documents:

https://github.com/krfricke/ray/blob/release/process/release/RELEASE_PROCESS.rst

https://github.com/krfricke/ray/blob/release/process/release/RELEASE_CHECKLIST.md

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -16,97 +16,52 @@ This checklist is meant to be used in conjunction with the RELEASE_PROCESS.rst d
...
+
+- [ ] ``weekly`` release test suite
+    - [ ] Test passing
 
 ## Final Steps
+- [ ] Check version strings once more
 - [ ] Anyscale Docker images built and deployed
 - [ ] ML Docker Image Updated
 - [ ] Wheels uploaded to Test PyPI
 - [ ] Wheel sanity checks with Test PyPI
```
Comment (simon-mo) [inline release/RELEASE_CHECKLIST.md:43]: add 3.9?

**Review by simon-mo — APPROVED** [review]

**Review by ericl — APPROVED** [review]
Awesome!

**Comment by ericl** [comment]
cc @jjyao if you have any questions reading through the new version

File: release/RELEASE_PROCESS.rst [hunk 2]
```diff
@@ -328,11 +314,57 @@ to proceed with the final stages of the release!
    
    Check the dockerhub to verify the update worked. https://hub.docker.com/repository/docker/rayproject/ray/tags?page=1&name=latest&ordering=last_updated
 
-10. **Send out an email announcing the release** to the employees@anyscale.com
-   Google group, and post a slack message in the Announcements channel of the
-   Ray slack (message a team lead if you do not have permissions.)
+10. **Release the Java packages to Maven**.
```
Comment (richardliaw) [inline release/RELEASE_PROCESS.rst:317]: Where should we be notifying the product team?

**Pull request merged**